data-pipeline-flink/assessment-aggregator/src/main/scala/org/sunbird/dp/assessment/task/AssessmentAggregatorStreamTask.scala

```diff
@@ -67,16 +68,21 @@ class AssessmentAggregatorStreamTask(config: AssessmentAggregatorConfig, kafkaCo
 
 // $COVERAGE-OFF$ Disabling scoverage as the below code can only be invoked within flink cluster
 object AssessmentAggregatorStreamTask {
-    def main(args: Array[String]): Unit = {
-        val configFilePath = Option(ParameterTool.fromArgs(args).get("config.file.path"))
-        val config = configFilePath.map {
-            path => ConfigFactory.parseFile(new File(path)).resolve()
-        }.getOrElse(ConfigFactory.load("assessment-aggregator.conf").withFallback(ConfigFactory.systemEnvironment()))
-        val assessmentAggregatorConfig = new AssessmentAggregatorConfig(config)
-        val kafkaUtil = new FlinkKafkaConnector(assessmentAggregatorConfig)
-        val task = new AssessmentAggregatorStreamTask(assessmentAggregatorConfig, kafkaUtil)
-        task.process()
-    }
+  def main(args: Array[String]): Unit = {
+    val configFilePath = Option(ParameterTool.fromArgs(args).get("config.file.path"))
+    val config = configFilePath.map {
+      path => ConfigFactory.parseFile(new File(path)).resolve()
+    }.getOrElse(ConfigFactory.load("assessment-aggregator.conf").withFallback(ConfigFactory.systemEnvironment()))
+    val assessmentAggregatorConfig = new AssessmentAggregatorConfig(config)
+    val kafkaUtil = new FlinkKafkaConnector(assessmentAggregatorConfig)
+    val task = new AssessmentAggregatorStreamTask(assessmentAggregatorConfig, kafkaUtil)
+    task.process()
+  }
 }
 
+class AssessmentAggregatorKeySelector(config: AssessmentAggregatorConfig) extends KeySelector[Event, Int] {
+  override def getKey(in: Event): Int = {
+    in.userId.hashCode % config.windowShards
```

Comment:
Will it improve the performance if we use the userId, courseId as this composite key is the partitioning key in Cassandra?